b2share/factory.py

```diff
@@ -1,192 +1,178 @@
-    })
-    if api.config.get('WSGI_PROXIES'):
-        wsgi_proxies = api.config.get('WSGI_PROXIES')
-        assert(wsgi_proxies > 0)
-        api.wsgi_app = ProxyFix(api.wsgi_app,
-                                num_proxies=api.config['WSGI_PROXIES'])
-
-    check_configuration(api.config, api.logger)
-
-    return api
-
-
-def add_routes(app_ui):
-    @app_ui.route('/')
-    def root():
-        return app_ui.send_static_file('index.html')
-
-    @app_ui.route('/help', defaults={'path': ''})
-    @app_ui.route('/help/', defaults={'path': ''})
-    @app_ui.route('/help/<path:path>')
-    def serve_help(path):
-        return app_ui.send_static_file('index.html')
-
-    @app_ui.route('/communities', defaults={'path': ''})
-    @app_ui.route('/communities/', defaults={'path': ''})
-    @app_ui.route('/communities/<path:path>')
-    def serve_communities(path):
-        return app_ui.send_static_file('index.html')
-
-    @app_ui.route('/user', defaults={'path': ''})
-    @app_ui.route('/user/', defaults={'path': ''})
-    @app_ui.route('/user/<path:path>')
-    def serve_user(path):
-        return app_ui.send_static_file('index.html')
-
-    @app_ui.route('/records', defaults={'path': ''})
-    @app_ui.route('/records/', defaults={'path': ''})
-    @app_ui.route('/records/<path:path>')
-    def serve_records(path):
-        return app_ui.send_static_file('index.html')
-
-    @app_ui.route('/search', defaults={'path': ''})
-    @app_ui.route('/search/', defaults={'path': ''})
-    @app_ui.route('/search/<path:path>')
-    def serve_search(path):
-        return app_ui.send_static_file('index.html')
 
+    invenio_config_loader(app, **kwargs_config)
+
+
+class TrustedHostsMixin(object):
+    """Mixin for reading trusted hosts from application config."""
+
+    @property
+    def trusted_hosts(self):
+        """Get list of trusted hosts."""
+        if current_app:
+            return current_app.config.get('APP_ALLOWED_HOSTS', None)
+
+
+def app_class():
+    """Create Flask application class.
+
+    Invenio-Files-REST needs to patch the Werkzeug form parsing in order to
+    support streaming large file uploads. This is done by subclassing the Flask
+    application class.
+    """
+    try:
+        pkg_resources.get_distribution('invenio-files-rest')
+        from invenio_files_rest.app import Flask as FlaskBase
+    except pkg_resources.DistributionNotFound:
+        from flask import Flask as FlaskBase
+
+    # Add Host header validation via APP_ALLOWED_HOSTS configuration variable.
+    class Request(TrustedHostsMixin, FlaskBase.request_class):
+        pass
+
+    class Flask(FlaskBase):
+        request_class = Request
+
+    return Flask
+
+
+create_api = create_app_factory(
+    'invenio',
+    config_loader=config_loader,
+    blueprint_entry_points=['invenio_base.api_blueprints'],
+    extension_entry_points=['invenio_base.api_apps'],
+    converter_entry_points=['invenio_base.api_converters'],
+    wsgi_factory=wsgi_proxyfix(),
```

Comment:
ProxyFix from werkzeug contrib (https://github.com/EUDAT-B2SHARE/b2share/blob/8f31a1172d6b21ba7d86a5fc4c907c9103099ab9/b2share/factory.py#L95)changed to invenio based proxyfix.